docs/painless/painless-guide/painless-execute-script.asciidoc

```diff
@@ -164,3 +164,329 @@ Response:
+    true
+  ]
+}
+--------------------------------------------------
++
+* date_field
++
+[source,console]
+----------------------------------------------------------------
+PUT /my-index-date
+{
+  "mappings": {
+    "properties": {
+      "d": {
+        "type": "date"
+      }
+    }
+  }
+}
+
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": "emit(doc['d'].value.toInstant().toEpochMilli());"
+  },
+  "context": "date_field",
+  "context_setup": {
+    "index": "my-index-date",
+    "document": {
+      "d": 2075885659000
+    }
+  }
+}
+----------------------------------------------------------------
++
+Response:
++
+[source,console-result]
+--------------------------------------------------
+{
+  "result": [
+    2075885659000
+  ]
+}
+--------------------------------------------------
++
+* double_field
++
+[source,console]
+----------------------------------------------------------------
+PUT /my-index-double
+{
+  "mappings": {
+    "properties": {
+      "d": {
+        "type": "double"
+      }
+    }
+  }
+}
+
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": "emit(doc['d'][0] + doc['d'][1])"
+  },
+  "context": "double_field",
+  "context_setup": {
+    "index": "my-index-double",
+    "document": {
+      "d": [2.0, 1.0]
+    }
+  }
+}
+----------------------------------------------------------------
++
+Response:
++
+[source,console-result]
+--------------------------------------------------
+{
+  "result": [ 3.0 ]
+}
+--------------------------------------------------
++
+* geo_point_field
++
+[source,console]
+----------------------------------------------------------------
+PUT /my-index-geo-point
+{
+  "mappings": {
+    "properties": {
+      "g": {
+        "type": "geo_point"
+      }
+    }
+  }
+}
+
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": "emit(doc['g'].value.lat + 1.2, doc['g'].value.lon - 2.3)"
+  },
+  "context": "geo_point_field",
+  "context_setup": {
+    "index": "my-index-geo-point",
+    "document": {
+      "g": "71.34,85.92"
+    }
+  }
+}
+----------------------------------------------------------------
++
+Response:
++
+[source,console-result]
+--------------------------------------------------
+{
+  "result": [
+    7434037857170656882
+  ]
+}
+--------------------------------------------------
++
+* ip_field
++
+[source,console]
+----------------------------------------------------------------
+PUT /my-index-ip
+{
+  "mappings": {
+    "properties": {
+      "i": {
+        "type": "ip"
+      }
+    }
+  }
+}
+
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": "emit(doc['i'].value)"
+  },
+  "context": "ip_field",
+  "context_setup": {
+    "index": "my-index-ip",
+    "document": {
+      "i": "127.0.0.1"
+    }
+  }
+}
+----------------------------------------------------------------
++
+Response:
++
+[source,console-result]
+--------------------------------------------------
+{
+  "result": [
+    "AAAAAAAAAAAAAP//fwAAAQ=="
```

Comment:
I think you should try and plug in the ip address formatter here to get the right formatting.

Reply:
Thank you for the discussion here -- created another PR to make the output make more sense for date, geo_point, and ip types.

https://github.com/elastic/elasticsearch/pull/72135